app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -174,11 +180,19 @@ class RoutePresenter(
                     "This is a developer bug, fixable by adding an action to graph_main.xml"
             )
         } else {
-            val clearBackStack = src.getAction(transition)?.navOptions?.shouldLaunchSingleTop() ?: false
+
+            val action = src.getAction(transition)
+            val clearBackStack = action?.navOptions?.shouldLaunchSingleTop() ?: false
             if (clearBackStack) {
                 while (navController.popBackStack()) {
                     // NOP
                 }
+                routeStore.clearBackStack()
+
+                action?.let {
```

Comment:
why is this inside of the `if (clearBackStack)` block?

Reply:
Yerk; you're right this could be cleaner.